s3_scrubber/src/checks.rs

```diff
@@ -178,69 +213,96 @@ pub(crate) struct S3TimelineBlobData {
 pub(crate) enum BlobDataParseResult {
     Parsed {
         index_part: IndexPart,
-        s3_layers: HashSet<LayerFileName>,
+        index_part_generation: Generation,
+        s3_layers: HashSet<(LayerFileName, Generation)>,
     },
     Incorrect(Vec<String>),
 }
 
+fn parse_layer_object_name(name: &str) -> Result<(LayerFileName, Generation), String> {
+    match name.rsplit_once('-') {
+        // FIXME: this is gross, just use a regex?
+        Some(gen) if gen.1.len() == 8 => {
```

Comment:
this would be nicer if it were `Some((name_prefix, name_suffix))` or something like that instead of using tuple accessors. Not everyone will be aware of what `rsplit_once` does precisely.

Reply:
fd90fa746